File: Data Structures & Algorithms/valid-palindrome-ii/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    validPalindrome(s) {

        const rec = (left, right, k) => {
            if (left >= right) return true
            if (s[left] === s[right]) return rec(left+1, right-1, k)
            if (k === 0) return false
            return rec(left+1, right, k-1) || rec(left, right-1, k-1)
        }
        return rec(0, s.length - 1, 1)
    }
}

File: Data Structures & Algorithms/range-sum-of-bst/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        
        ans = 0
        def rangeSumBSTHelper(root: Optional[TreeNode]):
            nonlocal ans
            if not root:
                return
            
            if low <= root.val <= high:
                ans += root.val
            
            rangeSumBSTHelper(root.left)
            rangeSumBSTHelper(root.right)
            
        rangeSumBSTHelper(root)
        return ans